src/dfx/src/commands/replica.rs

```diff
@@ -88,26 +89,32 @@ pub fn exec(env: &dyn Environment, opts: ReplicaOpts) -> DfxResult {
     let system = actix::System::new();
 
     let btc_adapter_pid_file_path = env.get_temp_dir().join("ic-btc-adapter-pid");
+    let btc_adapter_config_path = env.get_temp_dir().join("ic-btc-adapter-config.json");
+
     std::fs::write(&btc_adapter_pid_file_path, "")?;
 
+    let config = env.get_config_or_anyhow()?;
+    let btc_adapter_config = write_btc_adapter_config_if_enabled(
+        config.get_config(),
+        &btc_adapter_config_path,
+        opts.enable_bitcoin,
+        opts.bitcoin_node.clone(),
+    )?;
```

Comment:
```suggestion
    let btc_adapter_config = write_btc_adapter_config_if_enabled(
        config.get_config(),
        &btc_adapter_config_path,
        opts.enable_bitcoin,
        opts.bitcoin_node.clone(),
    ).context("Failed to write BTC adapter config.")?;
```
Could you add some more `context`s in the spirit of #2163 ? I think right now it could produce a simple `Premission denied` with no way to understand what's happening.

Reply:
Done, but at a lower level, since this context message here wouldn't actually be adding context at this level.